dbx/commands/sync.py

```diff
@@ -0,0 +1,384 @@
+):
+    """
+    Performs the initial sync from the source directory and then watches for changes, performing
+    an incremental sync whenever changes are detected.
+    """
+
+    matcher = create_path_matcher(source=source, includes=includes, excludes=excludes)
+
+    syncer = RemoteSyncer(
+        client=client,
+        source=source,
+        includes=includes,
+        excludes=excludes,
+        dry_run=dry_run,
+        full_sync=full_sync,
+        matcher=matcher,
+        delete_unmatched_option=delete_unmatched_option,
+    )
+
+    # Windows by default uses a different event loop policy which results in "Event loop is closed" errors
+    # for some reason.
+    if platform.system() == "Windows":
+        asyncio.set_event_loop_policy(asyncio.WindowsSelectorEventLoopPolicy())
+
+    dbx_echo("Starting initial copy")
+
+    # Run the incremental copy and record how many operations were performed or would have been
+    # performed (if in dry run mode).  An operation usually translates to an API call, such as
+    # create a directory, put a file, etc.
+    op_count = asyncio.run(syncer.incremental_copy())
+
+    if not op_count:
+        dbx_echo("No changes found during initial copy")
+
+    if dry_run:
+        dbx_echo("This was a dry run.  Exiting now.")
+    elif watch:
+        dbx_echo("Done. Watching for changes...")
+
+        with file_watcher(source=source, matcher=matcher) as event_handler:
+            while True:
+                while True:
+                    events = event_handler.get_events()
+                    if events:
+                        break
+                    time.sleep(sleep_interval)
+
+                try:
+                    op_count = asyncio.run(syncer.incremental_copy())
+                except StopIncrementalCopy:
+                    # simple way to enable unit testing to break out of loop
+                    break
+
+                if op_count > 0:
+                    dbx_echo("Done")
+
+
+def subdirs_to_patterns(source: str, subdirs: List[str]) -> List[str]:
+    """Converts a list of subdirectories under a source directory to a
+    list of gitignore-style patterns that match those directories.
+
+    Args:
+        source (str): source directory these subdirs exist under
+        subdirs (List[str]): subdirectories to create patterns for
+
+    Raises:
+        click.BadArgumentUsage: directory doesn't exist
+
+    Returns:
+        List[str]: list of patterns matching subdirs
+    """
+    patterns = []
+    for subdir in subdirs:
+        full_subdir = os.path.join(source, subdir)
+        if not os.path.exists(full_subdir):
+            raise click.BadArgumentUsage(f"Path {full_subdir} does not exist")
+        subdir = Path(subdir).as_posix()
+        patterns.append(f"/{subdir}/")
+    return patterns
+
+
+@cli.command()
+@click.option("--profile", type=str, help="The databricks-cli profile to use for accessing the API token")
+def check(profile: str):
```

Comment:
I found it useful to test whether my token worked without having to perform a sync.

Reply:
I've removed the `check` command because I think the token configuration seems to be less error prone now that I'm using classes from the Databricks CLI to get the config, compared to a previous method I was using before to manage the token.